eclipse-theia/theia
# Introduced config dir on the env variable server

Follow-up of #6650.

#### What it does
From now on, all configuration (`settings.json`, `keymaps.json`, etc) and VS Code extension things (such as `globalStorage`, `logs`, and `plugin-storage`) goes directly under the configuration folder.
Downstream projects can override the default, `.theia` folder name for the configurations. See the example [here](https://github.com/eclipse-theia/theia/commit/8ab1980777d3d673b5c64cc7e95dbb306412b71e).

What it does not do: 
 - we do not distinguish between cachable (`XDG_CACHE_HOME` on Linux and `AppData/Local` on Windows, such as workspace storage, extensions and their logs for example) and shareable things (`XDG_CONFIG_HOME` on Linux and `AppData/Roaming` on WIndows, such as `settings.json`, `keymaps.json`)


#### How to test

Try the example customization:
 - try to run a task from a workspace location (note, `tasks.json` should go under `.theia-example`),
 - see you have the `.theia-example` folder under `~/`,
 - POSIX: you can open recent workspaces and see if the paths are still correctly tildyfied.

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
